test/values.test.js

```diff
@@ -501,12 +538,178 @@ describe('Values', () => {
     // versioned queries/lambdas
     assertPrint(
       new Query({ api_version: '2.12', lambda: 'X', expr: { var: 'X' } }),
-      `Query(Lambda("X", Var("X")))`
+      'Query(Lambda("X", Var("X")))'
     )
 
     assertPrint(
       new Query({ api_version: '3', lambda: 'X', expr: { var: 'X' } }),
-      `Query(Lambda("X", Var("X")))`
+      'Query(Lambda("X", Var("X")))'
+    )
+
+    assertPrint(
+      new Query({ lambda: 'X', expr: { var: 'X' }, api_version: '3' }),
+      'Query(Lambda("X", Var("X")))'
+    )
+  })
+
+  test('pretty print despite argument order', () => {
+    // Match
+    assertPrint(
+      new Query({
+        api_version: '3',
+        expr: {
+          match: {
+            index: 'idx',
+          },
+          terms: ['str', 10],
+        },
+        lambda: '_',
+      }),
+      'Query(Lambda("_", Match(Index("idx"), ["str", 10])))'
+    )
+
+    assertPrint(
+      new Query({
+        expr: {
+          terms: 10,
+          match: {
+            index: 'idx',
+          },
+        },
+        api_version: '3',
+        lambda: '_',
+      }),
+      'Query(Lambda("_", Match(Index("idx"), 10)))'
+    )
+
+    // Filter an array
+    assertPrint(
+      new Query({
+        collection: [1, 2, 3],
+        api_version: '3',
+        filter: {
+          expr: {
+            equals: [0, { modulo: [{ var: 'i' }, 2] }],
+          },
+          lambda: 'i',
+        },
+      }),
+      'Query(Filter([1, 2, 3], Lambda("i", Equals(0, Modulo(Var("i"), 2)))))'
```

Comment:
although this is kinda acceptable for testing purposes, you will never receive a `@query` type with a `Filter` expression. Queries types only wrap Lambdas

Reply:
@marrony I noticed this when testing queries in the webshell that no `@query` was coming back for certain things, but my assertions were failing for some reason without `Query()`. I'm happy to go back and update these if you'd like 👍

Reply:
I don’t think it’s necessary to change, it was just a heads up to make sure you add/keep the lambdas tests

Reply:
In the future, would the best way to do it be using `new Expr()`?

Reply:
`Query` extends from `Expr`

Reply:
but it could probably be a good idea to start using `Expr` like those tests https://github.com/fauna/faunadb-js/pull/310/files#diff-04f5179b95b9dabc1a5d3bfa30234fffR716